pyirf/interpolation.py

```diff
@@ -2,22 +2,238 @@
+    )
+
+    return pdf_values
+
+
+def norm_pdf(pdf_values):
+    """
+    Normalize binned_pdf to a sum of 1
+    """
+    norm = np.sum(pdf_values, axis=-1)
+
+    # Ignore errors from rows without errors as they are set to 0 through nan_to_num
+    with np.errstate(invalid="ignore"):
+        normed_pdf_values = pdf_values / norm[..., np.newaxis]
+    return np.nan_to_num(normed_pdf_values)
+
+
+def interpolate_binned_pdf(edges, binned_pdf, m, mprime, axis, quantile_resolution):
+    """
+    Takes a grid of binned pdfs for a bunch of different parameters
+    and interpolates it to given value of those parameters.
+    This function provides an adapted version of the quantile interpolation introduced
+    in [1].
+    Instead of following this method closely, it implements different approaches to the
+    steps shown in Fig. 5 of [1].
+
+    Parameters
+    ----------
+    edges: numpy.ndarray, shape=(M+1)
+        Common array of bin-edges (along the abscissal ("x") axis) for the M bins of the input templates
+
+    binned_pdf: numpy.ndarray, shape=(N,...,M,...)
+        Array of M bin-heights (along the ordinate ("y") axis) for each of the N input templates.
+        The distributions to be interpolated (e.g. MigraEnerg for the IRFs Energy Dispersion) is expected to
+        be given at the dimension specified by axis.
+
+    m: numpy.ndarray, shape=(N, O)
+        Array of the N O-dimensional morphing parameter values corresponding to the N input templates. The pdf's quantiles
+        are expected to vary linearly between these two reference points.
+
+    m_prime: numpy.ndarray, shape=(O)
+        Value for which the interpolation is performed (target point)
+
+    axis: int
+        Axis along which the pdfs used for interpolation are located
+
+    quantile_resolution: float
+        Spacing between the quantiles that are computed in the interpolation. Defaults to 1e-3.
+
+    Returns
+    -------
+    f_new: numpy.ndarray, shape=(1,...,M,...)
+        Interpolated and binned pdf
+
+    References
+    ----------
+    .. [1] B. E. Hollister and A. T. Pang (2013). Interpolation of Non-Gaussian Probability Distributions
+           for Ensemble Visualization
+           https://engineering.ucsc.edu/sites/default/files/technical-reports/UCSC-SOE-13-13.pdf
+    """
+    # To have the needed axis always at the last index, as the number of indices is
+    # not safely propageted through the interpolation but the last element remains the last element
+    binned_pdf = np.swapaxes(binned_pdf, axis, -1)
+
+    # compute quantiles from quantile_resolution
+    quantiles = np.linspace(0, 1, int(np.round(1 / quantile_resolution, decimals=0)))
+
+    # compute bin-mids from bin-edges
+    bin_mids = bin_center(edges)
+
+    # compute cdf values
+    cdfs = cdf_values(binned_pdf)
+
+    # compute ppf values at interpolation point, determine quantile and interpolate quantiles steps of [1]
+    ppfs = ppf_values(cdfs, bin_mids, m, mprime, quantiles)
```

Comment:
With not doing the interpolation inside `ppf_values` above, this would very clearly show the structure of the algorithm here!

Reply:
```
ppfs = ppf_values(...)
interpolated_ppfs = griddate(...)
interpolated_pdfs = ppf_to_pdf(...)
...
```